packages/@glimmer/runtime/tests/attributes-test.ts

```diff
@@ -26,7 +28,13 @@ function commonSetup() {
 function render<T>(template: Template<T>, context = {}, view: PathReference<Opaque> = null) {
   self = new UpdatableReference(context);
   env.begin();
-  result = template.render(self, root, new TestDynamicScope());
+  let templateIterator = template.render(self, root, new TestDynamicScope());
+
+  do {
+    result = templateIterator.next();
```

Comment:
don't assign the iterator to a closure, this messes up escape analysis.  the `IteratorResult<RenderResult>` should be local var, the `RenderResult` is the closure